dot/rpc/modules/state_test.go

```diff
@@ -280,15 +282,29 @@ func TestStateModuleGetKeysPaged(t *testing.T) {
 	}
 }
 
-// Implement Tests once function is implemented
+// TestCall tests the state_call.
+// TODO: cover more cases in Zombienet tests
+// https://github.com/ChainSafe/gossamer/issues/3229
 func TestCall(t *testing.T) {
 	ctrl := gomock.NewController(t)
+	testHash := common.NewHash([]byte{0x01, 0x02})
+	rt := wasmer.NewTestInstance(t, runtime.WESTEND_RUNTIME_v0929)
 
 	mockNetworkAPI := mocks.NewMockNetworkAPI(ctrl)
 	mockStorageAPI := mocks.NewMockStorageAPI(ctrl)
-	sm := NewStateModule(mockNetworkAPI, mockStorageAPI, nil, nil)
+	mockBlockAPI := mocks.NewMockBlockAPI(ctrl)
+	mockBlockAPI.EXPECT().BestBlockHash().Return(testHash)
+	mockBlockAPI.EXPECT().GetRuntime(gomock.Any()).Return(rt, nil)
+
+	sm := NewStateModule(mockNetworkAPI, mockStorageAPI, nil, mockBlockAPI)
 
-	err := sm.Call(nil, nil, nil)
+	req := &StateCallRequest{
+		Method: "Core_version",
+		Params: "0x",
+		Block:  &testHash,
+	}
+	var res StateCallResponse
+	err := sm.Call(nil, req, &res)
```

Comment:
Do we want to assert on res here?

Reply:
Yes and it needs to cover all the usecases. It's being tracked here https://github.com/ChainSafe/gossamer/issues/3229

Reply:
Can we check at least for this single test case? Might be useful to copy paste expectations from here to zombienet tests 🤔

Reply:
Even if we leave as is, if its being tracked in a different PR could we add a comment with link to the issue tracking it?

Reply:
> Can we check at least for this single test case? Might be useful to copy paste expectations from here to zombienet tests 🤔
@qdm12 I'll be doing it the other way around

Reply:
> Even if we leave as is, if its being tracked in a different PR could we add a comment with link to the issue tracking it?

@jimjbrettj I've created another issue for this.